### PR TITLE
refactor(transactions): move TxEip1559, TxEip1559 and TxEip4844 to separate files

### LINKED ISSUE #3942: Transaction def module clean up
### Describe the feature

we're currently using a single file for all txs:

https://github.com/paradigmxyz/reth/blob/b5a44aeb3764265eb26b5c9794371710513f4875/crates/primitives/src/transaction/mod.rs

with new eip4844 blob tx this makes it even harder to navigate

## TODO
move all TxEip* structs to their respective modules and only keep `Transaction` in mod.rs

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by PanGan21**
Closes: #3942

### CODE REVIEW

File: crates/primitives/src/transaction/tx_access_list.rs [hunk 1]
```diff
@@ -0,0 +1,69 @@
+use crate::{Bytes, ChainId, TransactionKind};
+use std::mem;
+
```
Comment (mattsse) [inline crates/primitives/src/transaction/tx_access_list.rs:3]:
```suggestion
```

File: crates/primitives/src/transaction/tx_access_list.rs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+use reth_codecs::{main_codec, Compact};
+
```
Comment (mattsse) [inline crates/primitives/src/transaction/tx_access_list.rs:5]:
```suggestion
```

**Review by mattsse — CHANGES_REQUESTED** [review]
nice,

I'd like to rename files

* legacy
* eip2930
* eip1559
* eip4844

and also move new TxEip4844 to new file

**Comment by PanGan21** [comment]
> nice,
> 
> I'd like to rename files
> 
> * legacy
> * eip2930
> * eip1559
> * eip4844
> 
> and also move new TxEip4844 to new file

Should be fixed! Thanks for the suggestions.
I am wondering if I it is better to also move the unit tests or keep them in `mod.rs`

**Comment by mattsse** [comment]
> I am wondering if I it is better to also move the unit tests or keep them in mod.rs

very good point, let's do that as well!

we now have a smol conflict with affects the effective gas price function, and some comments

https://github.com/paradigmxyz/reth/pull/3943

File: crates/primitives/src/transaction/tx_access_list.rs (hunk 1 above)
Comment (PanGan21) [reply crates/primitives/src/transaction/tx_access_list.rs]: fixed

File: crates/primitives/src/transaction/tx_access_list.rs (hunk 2 above)
Comment (PanGan21) [reply crates/primitives/src/transaction/tx_access_list.rs]: fixed

**Comment by PanGan21** [comment]
> > I am wondering if I it is better to also move the unit tests or keep them in mod.rs
> 
> very good point, let's do that as well!
> 
> we now have a smol conflict with affects the effective gas price function, and some comments
> 
> #3943

The refactor should be done. I kept the `decode_multiple_network_txs` in the same file in order to not duplicate the helper `test_decode_empty_typed_tx`. Let me know if something more is required!

**Review by mattsse — APPROVED** [review]
awesome, tysm for this

**Pull request merged**